collector/container/konflux-slim.Dockerfile

```
@@ -1,55 +1,51 @@
 ARG BUILD_DIR=/build
 ARG CMAKE_BUILD_DIR=${BUILD_DIR}/cmake-build
 
+# Builder
 # TODO(ROX-20312): we can't pin image tag or digest because currently there's no mechanism to auto-update that.
 # TODO(ROX-20651): Use RHEL/ubi base image when entitlement is solved.
 # RPMs requiring entitlement: bpftool, cmake-3.18.2-9.el8, elfutils-libelf-devel, tbb-devel, c-ares-devel, jq-devel
 # FROM registry.access.redhat.com/ubi8/ubi:latest AS builder
-FROM quay.io/centos/centos:stream8 AS builder
+FROM registry.access.redhat.com/ubi8/ubi:latest AS ubi-normal
+FROM registry.access.redhat.com/ubi8/ubi:latest AS rpm-implanter-builder
 
-COPY . .
+COPY --from=ubi-normal / /mnt
+COPY ./.rhtap /tmp/.rhtap
 
 # TODO(ROX-20234): use hermetic builds when installing/updating RPMs becomes hermetic.
-RUN dnf upgrade -y --nobest \
-    && dnf -y install --nobest \
-        autoconf \
-        automake \
-        binutils-devel \
-        bison \
-        ca-certificates \
-        clang-15.0.7 \
-        cmake \
-        cracklib-dicts \
-        diffutils \
-        elfutils-libelf-devel \
-        file \
-        flex \
-        gcc \
-        gcc-c++ \
-        gdb \
-        gettext \
-        git \
-        glibc-devel \
-        libasan \
-        libubsan \
-        libcap-ng-devel \
-        libcurl-devel \
-        libtool \
-        libuuid-devel \
+RUN /tmp/.rhtap/scripts/subscription-manager-bro.sh register /mnt && \
+    dnf -y --installroot=/mnt upgrade --nobest && \
+    dnf -y --installroot=/mnt install --nobest \
         make \
-        openssh-server \
-        openssl-devel \
-        patchutils \
-        passwd \
-        pkgconfig \
-        rsync \
-        tar \
-        unzip \
-        valgrind \
         wget \
-        which \
+        unzip \
+        clang \
         bpftool \
-    && dnf clean all
+        cmake-3.18.2-9.el8 \
+        gcc-c++ \
+        openssl-devel \
+        ncurses-devel \
+        curl-devel \
+        libuuid-devel \
+        libcap-ng-devel \
+        autoconf \
+        libtool \
+        git \
+        elfutils-libelf-devel \
+        tbb-devel \
+        jq-devel \
+        c-ares-devel && \
+    /tmp/.rhtap/scripts/subscription-manager-bro.sh cleanup && \
+    # We can do usual cleanup while we're here: remove packages that would trigger violations. \
+    dnf -y --installroot=/mnt clean all && \
+    rpm --root=/mnt --verbose -e --nodeps $(rpm --root=/mnt -qa 'curl' '*rpm*' '*dnf*' '*libsolv*' '*hawkey*' 'yum*') && \
+    rm -rf /mnt/var/cache/dnf /mnt/var/cache/yum
```

Removing package managers isn't necessary from the builder stage, only the final stage. This removal is done to satisfy ACS own policy about having package manager in the image which was requested for some compliance policies.

```suggestion
    dnf -y --installroot=/mnt clean all
```

---

https://github.com/stackrox/collector/pull/1573/commits/57288765f6b72593489447a18a1f6694ca9e94a8#diff-ae246d60ffd0ce9d7455810ae6cdb168e932b1285a90712fbae086b826b3f501R39